### PR TITLE
Unit testing for the Logistic Predictor class. Fixes issue #38.

### PR DESCRIPTION
**Pull request by Dimkoim**

### CODE REVIEW

File: toxicity/test/test_linear_predictor.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import numbers
+import unittest
+import pathmagic  # noqa
```
Comment (steremma) [inline toxicity/test/test_linear_predictor.py]: Is #noqa needed here? If yes what is the purpose of pathmagic?

File: toxicity/test/test_linear_predictor.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import pandas as pd
+from preprocessing import tf_idf
+
+train_file = "../data/train.csv"
+test_file = "../data/test.csv"
+
+
+class TestLinearPredictor(unittest.TestCase):
+    # We will use a chunk of the dataset of n rows
+    number_of_rows = 1000
```
Comment (steremma) [inline toxicity/test/test_linear_predictor.py]: I dont think we need this here, just do `self.train = pd.read_csv(train_file, nrows=1000)` in the SetUp

File: toxicity/test/test_linear_predictor.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    # Logistic predictor parameters
+    lr_params = {"C": 4, "dual": True}
```
Comment (steremma) [inline toxicity/test/test_linear_predictor.py]: Same as above

File: toxicity/test/test_linear_predictor.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+    def setUp(self, tf_idf_func=True):
+        self.train = pd.read_csv(train_file, nrows=TestLinearPredictor.number_of_rows)
+        self.test = pd.read_csv(test_file, nrows=TestLinearPredictor.number_of_rows)
+        self.y_train = {tag: self.train[tag].values for tag in utils.TAGS}
+        self.logistic_predictor = LogisticPredictor(**TestLinearPredictor.lr_params)
+        if tf_idf_func:
```
Comment (steremma) [inline toxicity/test/test_linear_predictor.py]: Does it ever make sense to do anything without `tf_idf`? I propose to remove the argument and always call tf_idf

File: toxicity/test/test_linear_predictor.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+    def test_predict_proba(self):
+        """We will test for the 'toxic' tag"""
+        self.logistic_predictor.fit(self.train, self.y_train['toxic'])
+        predictions = self.logistic_predictor.predict(self.test)
+        assert predictions.shape == (self.test.shape[0],)
+        assert (predictions <= 1).all() and (predictions >= 0).all()
+
+    @staticmethod
+    def idf(train_df, test_df):
```
Comment (steremma) [inline toxicity/test/test_linear_predictor.py]: Delete this, directly call tf_idf

File: toxicity/test/test_linear_predictor.py (hunk 1 above)
Comment (Dimkoim) [reply toxicity/test/test_linear_predictor.py]: yes, it is needed in order to be compliant with the PEP8.

File: toxicity/test/test_linear_predictor.py (hunk 2 above)
Comment (Dimkoim) [reply toxicity/test/test_linear_predictor.py]: I will leave in order to be things clear.

File: toxicity/test/test_linear_predictor.py (hunk 4 above)
Comment (Dimkoim) [reply toxicity/test/test_linear_predictor.py]: You are right!

File: toxicity/test/test_linear_predictor.py (hunk 5 above)
Comment (Dimkoim) [reply toxicity/test/test_linear_predictor.py]: Indeed.

**Pull request merged**